nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -311,17 +315,27 @@ in
         neededForBoot = true;
       };
 
-    fileSystems."/nix/store" =
+    fileSystems."/nix/store" = if config.isoImage.unionfs
+      then
       { fsType = "unionfs-fuse";
         device = "unionfs";
         options = [ "allow_other" "cow" "nonempty" "chroot=/mnt-root" "max_files=32768" "hide_meta_files" "dirs=/nix/.rw-store=rw:/nix/.ro-store=ro" ];
+      }
+      else
+      { fsType = "overlay";
+        device = "overlay";
+        options = [
+          "lowerdir=/mnt-root/nix/.ro-store"
```

Comment:
This interface really shouldn't depend on the detail that the initrd mounts the root on `/mnt-root`, because it would make it inconsistent when specifying overlayfs mounts that should _not_ get mounted in initrd.

Instead, I suppose the bash code below that creates the directories should prepend `/mnt-root` at runtime.

Reply:
The thing is this need to be passed to mount options like so as there's no chroot option.